papers/pydra/paper.rst

```diff
@@ -108,42 +114,54 @@ or for use in applications where caching and distributed execution are not neces
 Since Pydra relies on a filesystem cache at present it is also not
 designed for dataflows that need to operate purely in memory. 
 
-
 The next section will describe the Pydra architecture --- main package classes
 and interactions between them. The *Key Features* section focuses on a set of features whose
 combination distinguishes Pydra from other dataflow engines. The paper concludes with a set
 of applied examples demonstrating the power and utility of Pydra.
 
 
-TODO: provenance??
-
 Architecture
 ------------
-*Pydra* architecture has three main components |---| runnable objects, i.e. *Tasks*,
-*Submitter* and *Worker*.
-There is one type of *Submitter*, but several types of *Tasks*
-and *Workers* have been developed, see schematic presentation in Fig. :ref:`classes`.
-In the following subsection all of these components will be briefly described.
-
-
+*Pydra* architecture has three core components: *Task*, *Submitter* and *Worker*.
+*Tasks* form the basic building blocks of the dataflow, while *Submitter*
+orchestrates the dataflow execution model. Different types of *Workers* allow
+Pydra to execute the task on different compute architectures. Fig. :ref:`classes`
+shows the Class hierarchy and links between them in the present Pydra
+architecture. In order to describe Pydra's most notable features in the next
+section, we briefly describe the role and function of each of these classes.
 
 .. figure:: classes.pdf
    :figclass: h!
    :scale: 30%
 
    A schematic presentation of principal classes in Pydra. :label:`classes`
 
-
 Dataflows Components: Task and Workflow
 =======================================
-A *Task* is the basic runnable component of *Pydra* and is descibed by the class ``TaskBase``.
-There are several classes that inherit from ``TaskBase`` and each has a different application:
+A *Task* is the basic runnable component of *Pydra* and is described by the
+class ``TaskBase``. A *Task* has named inputs and outputs thus allowing
+construction of dataflows. It can be hashed and executes in a specific working
+directory. There are several classes that inherit from ``TaskBase`` and each has
+a different application:
 
-* ``FunctionTask`` is a *Task* that is design as a wrapper for Python function.
-  Every Python defined function could be transformed to the ``FunctionTask`` by using Pydra
-  decorator - ``mark.task``.
-  In addition, the user can use Python's function annotation or another Pydra decorator
-  |---| ``mark.annotate`` in order to specify the output, see an example below:
+* ``FunctionTask`` is a *Task* that executes Python functions. Most Python functions
+  declared in an existing library, package, or interactively can be converted to
```

Comment:
i'm not sure what you mean in this line

Reply:
it means we can turn existing functions into pydra function tasks?

Reply:
yes, but wasn't sure about this "r interactively", etc., but we can keep it